controllers/provisioners/eks/helpers.go

```diff
@@ -1173,3 +1173,40 @@ func (ctx *EksInstanceGroupContext) GetDesiredMixedInstancesPolicy(name string)
 
 	return policy
 }
+
+func filterSupportedArch(architectures []*string) string {
```

Comment:
Once you return StringValueSlice in the previous call to get architecture, you can refactor this function a bit.
e.g., you can simply use ContainsEqualFold from here:
https://github.com/keikoproj/instance-manager/blob/c0eb25d65856ca1261f70465c56950f4bfc0cdc3/controllers/common/utils.go#L80

You can also make `SupportedArchitectures` available at a package level in case it needs to be used elsewhere